packages/middleware/src/PolyMiddlewareComposer.tsx

```diff
@@ -0,0 +1,43 @@
+import { reactNode, validateProps } from 'botframework-webchat-react-valibot';
+import React, { memo, useMemo } from 'react';
+import {
+  array,
+  custom,
+  function_,
+  object,
+  optional,
+  pipe,
+  readonly,
+  safeParse,
+  transform,
+  type InferInput
+} from 'valibot';
+
+import { ActivityPolyMiddlewareProvider, extractActivityPolyMiddleware } from './activityPolyMiddleware';
+import { PolyMiddleware } from './types/PolyMiddleware';
+
+const polyMiddlewareComposerPropsSchema = pipe(
+  object({
+    children: optional(reactNode()),
+    middleware: pipe(
+      custom<readonly PolyMiddleware[]>(value => safeParse(array(function_()), value).success),
+      transform<readonly PolyMiddleware[], readonly PolyMiddleware[]>(value => Object.freeze(Array.from(value)))
+    )
+  }),
+  readonly()
+);
+
+type PolyMiddlewareComposerProps = Readonly<InferInput<typeof polyMiddlewareComposerPropsSchema>>;
+
+function PolyMiddlewareComposer(props: PolyMiddlewareComposerProps) {
+  const { children, middleware } = validateProps(polyMiddlewareComposerPropsSchema, props);
+
+  const activityPolyMiddleware = useMemo(() => extractActivityPolyMiddleware(middleware), [middleware]);
+
+  return (
+    <ActivityPolyMiddlewareProvider middleware={activityPolyMiddleware}>{children}</ActivityPolyMiddlewareProvider>
```

Comment:
Doesn't it make sense to split poly middleware provider from the RCoR provider? This way we could have all middleware in a single provider until we get to the proxy component rendering. We'll need to wrap RCoR Proxy into the RCoR provider, but this way we get leaner tree in general.